### PR TITLE
add chunking api interface to management canister

### PR DESCRIPTION
**Pull request by sudoshreyansh**

### CODE REVIEW

**Comment by lastmjs** [comment]
Please follow these instructions: https://github.com/demergent-labs/azle?tab=readme-ov-file#contributing

**Comment by lastmjs** [comment]
If you are able to sign the license extension, I would like to ask for tests to be added to this PR as well. Most of our management canister tests are here: https://github.com/demergent-labs/azle/tree/main/examples/management_canister

You would need to add whatever extra functionality you've added to the management canister to this example similar to the other APIs in there, and add the tests as well.

That's all basic feedback before a thorough review.

**Comment by lastmjs** [comment]
Great I have merged the license extension signature and am now running the tests. Will review in the next few days.

File: canisters/management/canister_management.ts [hunk 1]
```diff
@@ -89,6 +94,28 @@ export const UpdateSettingsArgs = Record({
 });
 export type UpdateSettingsArgs = typeof UpdateSettingsArgs.tsType;
 
+export const UploadChunkArgs = Record({
+    canister_id: CanisterId,
```
Comment (lastmjs) [inline canisters/management/canister_management.ts:98]: Please match the `ic.did` file exactly and make `CanisterId` `Principal` for consistency.

**Review by lastmjs — CHANGES_REQUESTED** [review]

**Comment by lastmjs** [comment]
I've reviewed everything but the tests themselves, looking good so far just a few basic changes requested.

File: examples/management_canister/test/tests.ts [hunk 2]
```diff
@@ -61,49 +62,132 @@ export function getTests(managementCanister: ActorSubclass<_SERVICE>): Test[] {
...
-                const statusAfter = await managementCanister.getCanisterStatus({
-                    canister_id: canisterId
-                });
+                const wasmModule = Array.from(readFileSync('src/test.wasm'));
 
-                const cyclesAfter = statusAfter.cycles;
+                const chunkUploadResult =
+                    await managementCanister.executeUploadChunk(
+                        canisterId,
+                        wasmModule as any
```
Comment (lastmjs) [inline examples/management_canister/test/tests.ts:95]: Is the `as any` necessary? It isn't required on my machine at least.

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/management_canister/src/index.ts [hunk 3]
```diff
@@ -79,6 +82,84 @@ export default Canister({
...
+                        {
+                            canister_id: canisterId,
+                            chunk: chunk
+                        }
+                    ]
+                });
+            }
+        }
+    ),
+    executeClearChunkStore: update([Principal], bool, async (canisterId) => {
```
[the remark this answers is not in the document]
Comment (lastmjs) [reply examples/management_canister/src/index.ts:114]: Yep, the tests were great

**Comment by lastmjs** [comment]
I've finished my review, requesting a few changes, very impressed and thanks for the help!

**Comment by lastmjs** [comment]
Actually, there is another problem. You can ignore the other tests, but the bitcoin tests are failing and I don't believe they should be. It probably has something to do with updating the ic.did file. If you could address that so they are passing it would be great.

File: canisters/management/canister_management.ts (hunk 1 above)
Comment (sudoshreyansh) [reply canisters/management/canister_management.ts:98]: Yeah, I'll create an issue for this.

**Comment by sudoshreyansh** [comment]
Yeah, @lastmjs you were right regarding the bitcoin tests. Fixed it.
Also addressed the other requested changes.

**Comment by sudoshreyansh** [comment]
Hey, @lastmjs can you re-review when you have time?

**Comment by lastmjs** [comment]
Sorry for taking so long, will be reviewing soon. You couldn't have known this, but FYI we usually let the reviewer resolve conversations as part of the review process.

File: canisters/management/canister_management.ts (hunk 1 above)
Comment (lastmjs) [reply canisters/management/canister_management.ts:98]: Can you link to that issue here?

**Review by lastmjs — APPROVED** [review]

**Comment by lastmjs** [comment]
Everything is looking good, I've updated the branch and will just let the tests run one more time. If all looks good I'll be merging soon.

**Review by bdemann — APPROVED** [review]
LGTM

**Pull request merged**